aikau/src/main/resources/alfresco/documentlibrary/AlfCreateTemplateContentMenu.js

```diff
@@ -281,25 +288,50 @@ define(["dojo/_base/declare",
        * @param {object} widget The menu item to add
        * @param {integer} index The index to add the menu item at.
        */
-      _addMenuItem: function alf_menus_documentlibrary_AlfCreateTemplateContentMenu___addMenuItem(group, widget, index) {
+      _addMenuItem: function alfresco_documentlibrary_AlfCreateTemplateContentMenu___addMenuItem(group, widget, /*jshint unused:false*/ index) {
          var label = (widget.title !== "") ? widget.title : widget.name;
          var item = new AlfMenuItem({
             label: label,
             iconClass: this.templateIconClass,
-            publishTopic: this.templatePublishTopic,
+            publishTopic: this._menuItemToParentTopic,
             publishPayload: {
-               type: "template",
-               params: {
-                  sourceNodeRef: widget.nodeRef,
-                  targetNodeRef: this.targetNodeRef,
-                  templateType: this.templateType,
-                  name: widget.name,
-                  title: widget.title,
-                  description: widget.description
+               action: {
+                  type: "template",
+                  params: {
+                     sourceNodeRef: widget.nodeRef,
+                     targetNodeRef: null,
+                     templateType: this.templateType,
+                     name: widget.name,
+                     title: widget.title,
+                     description: widget.description
+                  }
                }
-            }
+            },
+            publishGlobal: true
+
          });
          group.addChild(item);
+      },
+
+      /**
+       * Adds the current node in which to create the templated content to the received payload and then publishes
+       * on to the [ActionService]{@link module:alfresco/services/ActionService} to create the templated content.
+       *
+       * @instance
+       * @param {object} payload The payload relating to the template to create.
+       */
+      onCreateTemplatedContent: function alfresco_documentlibrary_AlfCreateTemplateContentMenu__onCreateTemplatedContent(payload) {
+         var targetNodeRef = lang.getObject("currentNode.parent.nodeRef", false, this);
+         if (targetNodeRef)
+         {
+            payload.action.params.targetNodeRef = this.currentNode.parent.nodeRef;
+            payload.responseScope = this.pubSubScope;
+            this.alfPublish(this.templatePublishTopic, payload, true);
```

Comment:
Why not the following?

```
payload.action.params.targetNodeRef = this.currentNode.parent.nodeRef;
this.alfServicePublish(this.templatePublishTopic, payload);
```

Don't need to set responseScope (as it will set `this.pubSubScope` by default) and it is a service publish (I'm presuming) so would be good to start using the new method.